store/postgres/user_repository.go

```diff
@@ -16,14 +17,42 @@ type UserRepository struct {
 
 // Create insert a user to the database
 func (r *UserRepository) Create(ctx context.Context, ud *user.User) (string, error) {
+	return r.create(ctx, r.client.db, ud)
+}
+
+// Create insert a user to the database using given transaction as client
+func (r *UserRepository) CreateWithTx(ctx context.Context, tx *sqlx.Tx, ud *user.User) (string, error) {
```

Comment:
Our postgres client has a method `RunWithTx`. It is in `postgres.go`, maybe we can use that instead?

Reply:
oh I just saw this is actually a different kind of create. I thought this has several queries inside. I think it is fine for this.